### PR TITLE
fix issues, creates re-usable components

### PR DESCRIPTION
**Pull request by shakilhossain1**

### CODE REVIEW

File: components/location_service/location_service.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# In a file called [project root]/components/calendar/calendar.py
+from django_components import component
+
+@component.register("location_service")
+class Calendar(component.Component):
```
Comment (jbeard4) [inline components/location_service/location_service.py]: This class name of this component probably shouldn't be called `Calendar`

File: components/location_service/location_service.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    # Templates inside `[your apps]/components` dir and `[project root]/components` dir will be automatically found. To customize which template to use based on context
+    # you can override def get_template_name() instead of specifying the below variable.
+    template_name = "location_service/template.html"
+
+    # This component takes one parameter, a service_info string to show in the template
```
Comment (jbeard4) [inline components/location_service/location_service.py]: This comment seems inaccurate. It looks like this component takes 3 parameters.

Comment (jbeard4) [reply components/location_service/location_service.py]: Also, service_info is not a string, it's an object.

File: config/settings.py [hunk 3]
```diff
@@ -170,7 +178,7 @@
 
 STATIC_URL = "static/"
 STATIC_ROOT = 'static_dist'
-STATICFILES_DIRS = [str(BASE_DIR.joinpath("static"))]
+STATICFILES_DIRS = [str(BASE_DIR.joinpath("static")), os.path.join(BASE_DIR, "components"),]
```
Comment (jbeard4) [inline config/settings.py]: I don't think the components directory is meant to be served as static file.
```suggestion
STATICFILES_DIRS = [str(BASE_DIR.joinpath("static"))]
```

File: templates/location_details.html [hunk 4]
```diff
@@ -346,958 +347,36 @@
...
-                  </div>
-                </div>
-                {% endif %}
-              </div>
-            </div>
-          {% endfor %}
-
-        </div>
-      </div>
+      {% component "location_service" service_info=location.personal_care_services name="Personal care" icon="dist/img/icons/personal-care-icon.png" %}{% endcomponent %}
```
Comment (jbeard4) [inline templates/location_details.html:354]: This is a good cleanup. Lots off duplicate code removed.

**Review by jbeard4 — CHANGES_REQUESTED** [review]

File: config/settings.py [hunk 5]
```diff
@@ -178,7 +178,7 @@
 
 STATIC_URL = "static/"
 STATIC_ROOT = 'static_dist'
-STATICFILES_DIRS = [str(BASE_DIR.joinpath("static"))]
+STATICFILES_DIRS = [str(BASE_DIR.joinpath("static")), os.path.join(BASE_DIR, "components")]
```
Comment (jbeard4) [inline config/settings.py:181]: @shakilhossain1  Why was this restored?

Comment (shakilhossain1) [reply config/settings.py:181]: otherwise the django-component package won't working!

**Pull request merged**